src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioDocumentTrackerSubscriber.cs

```diff
@@ -0,0 +1,304 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Projection;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    [System.Composition.Shared]
+    [Export(typeof(VisualStudioDocumentTrackerSubscriber))]
+    internal class DefaultVisualStudioDocumentTrackerSubscriber : VisualStudioDocumentTrackerSubscriber
```

Comment:
Do we really need this to be a separate thing? It seems like this is a lot more complicated because it's separate.

Reply:
Looking through this more, it really seems like there's a lot more ceremony here than necessary. IMO this should all be part of the document tracker and nothing else.

Reply:
I'm saying it should be part of the VisualStudioDocumentTracker class.